From 789bc909ea59671e43fa89e539cca3b8970275fb Mon Sep 17 00:00:00 2001
From: Muna Sinada <msinada@codeaurora.org>
Date: Mon, 30 Nov 2020 20:37:14 -0800
Subject: [PATCH] hostapd: Initilize chan for second 80 mhz to zero

CSA updates freq for CSA beacons and after CSA beacons utilizing
hostapd_change_config_freq(). In the case of 80+80 MHz, there are two
channels store, one for the 80 MHz and the second 80 MHz. In non
80+80 cases, the second channel needs to be initialized to zero or
else garbage values are stored in place.

Signed-off-by: Muna Sinada <msinada@codeaurora.org>
---
 src/ap/hostapd.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/ap/hostapd.c b/src/ap/hostapd.c
index 346d83cfda91..fa46667017f4 100644
--- a/src/ap/hostapd.c
+++ b/src/ap/hostapd.c
@@ -3455,7 +3455,7 @@ static int hostapd_change_config_freq(struct hostapd_data *hapd,
 				      struct hostapd_freq_params *old_params)
 {
 	int channel;
-	u8 seg0, seg1;
+	u8 seg0, seg1 = 0;
 	struct hostapd_hw_modes *mode;
 
 	if (!params->channel) {
-- 
2.7.4

